src/ExportEdiJobs/ExportEdiJobDetailsActionMenu/ExportEdiJobDetailsActionMenu.test.js

```diff
@@ -64,4 +82,45 @@ describe('ExportEdiJobDetailsActionMenu', () => {
 
     expect(scheduleExportJob).toHaveBeenCalled();
   });
+
+  it('should handle export job download (even if status failed, but export file exists)', async () => {
+    renderExportEdiJobDetailsActionMenu({
+      exportJob: {
+        status: EXPORT_JOB_STATUSES.failed,
+        fileNames: ['test.edi'],
+      }, 
+    });
+
+    await act(async () => user.click(screen.getByTestId('job-action-download')));
+
+    expect(kyGetMock).toHaveBeenCalled();
+  });
+
+  it('should handle export job download error', async () => {
+    kyGetMock.mockReturnValue({ blob: () => Promise.reject() });
+
+    renderExportEdiJobDetailsActionMenu();
+
+    await act(async () => user.click(screen.getByTestId('job-action-download')));
+
+    expect(showCallout).toHaveBeenCalled();
```

Comment:
Maybe it's also worth checking that it was called with correct message? And same for re-send job

Reply:
Yeah, makes sense, thanks